src/main/java/org/jenkinsci/plugins/workflow/support/pickles/serialization/RiverReader.java

```diff
@@ -164,9 +168,12 @@ public Unmarshaller apply(PickleResolver input) {
         }
     }
 
+    @SuppressWarnings("unchecked")
     private List<Pickle> readPickles(int offset) throws IOException {
-        BufferedInputStream es = openStreamAt(offset);
-        try {
+        if (offset == 0) {
+            throw new IOException("No offset for pickles");
```

Comment:
Observed this error in a controller I am testing, so the code in the patch works as expected. Unfortunately I was not able to locate a log file covering the time period of the particular build with the error so I am not sure yet whether the fine logging added to `RiverWriter` will be helpful in tracking down the root cause; I do see these three messages being printed in the expected order for other builds.